benchmarks/cpp/src/qsort.cpp

```diff
@@ -0,0 +1,84 @@
+#include <asio.hpp>
+#include <iostream>
+#include <vector>
+#include <algorithm>
+#include <numeric>
+#include <random>
+
+using namespace asio;
+
+// A function to perform quicksort on a vector of integers
+void quicksort(std::vector<int>& vec, int low, int high)
+{
+    if (low < high)
+    {
+        int pivot = low;
+        int left = low + 1;
+        int right = high;
+        while (left <= right)
+        {
+            while (left <= right && vec[left] <= vec[pivot])
+            {
+                left++;
+            }
+            while (left <= right && vec[right] > vec[pivot])
+            {
+                right--;
+            }
+            if (left < right)
+            {
+                std::swap(vec[left], vec[right]);
+            }
+        }
+        std::swap(vec[pivot], vec[right]);
+        int pivot_point = right;
+        quicksort(vec, low, pivot_point - 1);
+        quicksort(vec, pivot_point + 1, high);
```

Comment:
The quicksort isn't executed concurrently, so this isn't a fair comparison of ASIO vs the other thread pool variants.

Reply:
Sorry, what do you suggest to fix this?

Reply:
Im not too familiar with ASIO, but it should schedule at least one of the quicksort's dynamically to the other and wait for both to complete. The golang example showcases this more clearly: https://github.com/kprotty/zap/blob/blog/benchmarks/go/qsort.go#L58-L68

Reply:
OK. I'll review the quicksort implementation.